kqueen_ui/blueprints/ui/templates/ui/partial/selectform.html

```diff
@@ -30,59 +30,60 @@
 {% endif %}
 {% if tagged_fields|length > 0 %}
 <script>
-function validateJsonFileField(value, element, requiredKeys) {
-  var file = element.files[0],
-      reader = new FileReader(),
-      valid = false;
-
-  reader.onload = function(evt) {
-    var data = evt.target.result;
-    var json = 'error';
-    var message = 'Validation pending'
-    try {
-      json = JSON.parse(data);
-    } catch(err) {
-      message = 'Not a valid JSON'
-      console.log(err);
-    }
-    if (json != 'error') {
-      var missingKeys = [];
-      var countKeys = 0;
-      requiredKeys.forEach(function(key) {
-        contains = Object.keys(json).indexOf(key) != -1;
-        if ( contains ) {
-          countKeys++;
-        } else {
-          missingKeys.push(key);
-        }
-      });
-      if ( countKeys != requiredKeys.length ) {
-        message = 'Missing mandatory keys: ' + missingKeys.toString();
+function validateFileField(element, requiredKeys, parser, validationFailedMessage) {
+  var data = 'error',
+      message = 'Validation failed',
+      isValid = false;
+  try {
+    data = parser(element.value);
+  } catch(err) {
+    message = validationFailedMessage;
```

Comment:
can we passed error from err? It will indicates where exactly file is corrupt and user can fix it right away? (number of wrong string and symbol should be provided)

Reply:
Done